clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/BackgroundEventHandler.java

```diff
@@ -42,6 +51,8 @@ public BackgroundEventHandler(final Queue<BackgroundEvent> backgroundEventQueue)
      */
     public void add(BackgroundEvent event) {
         Objects.requireNonNull(event, "BackgroundEvent provided to add must be non-null");
+        event.setAddedToQueueMs(System.currentTimeMillis());
```

Comment:
Same comment here: we need to update the constructor to provide a `Time` object and then use that here.